hudi-common/src/main/java/org/apache/hudi/common/table/read/HoodieKeyBasedFileGroupRecordBuffer.java

```diff
@@ -108,10 +114,20 @@ public void processDeleteBlock(HoodieDeleteBlock deleteBlock) throws IOException
   @Override
   public void processNextDeletedRecord(DeleteRecord deleteRecord, Serializable recordKey) {
     Pair<Option<T>, Map<String, Object>> existingRecordMetadataPair = records.get(recordKey);
+    if (deleteRecord.getOrderingValue() == null && existingRecordMetadataPair != null) {
+      existingRecordMetadataPair.getRight().put(DELETE_FOUND_WITHOUT_ORDERING_VALUE, "true");
+      return;
+    }
+
     Option<DeleteRecord> recordOpt = doProcessNextDeletedRecord(deleteRecord, existingRecordMetadataPair);
     if (recordOpt.isPresent()) {
+      Comparable orderingVal = recordOpt.get().getOrderingValue() == null ? orderingFieldDefault : recordOpt.get().getOrderingValue();
```

Comment:
we do this a couple times can we make a util function to do this?

Reply:
Added into DeleteRecord.